src/server/game/Chat/ChatCommands/ChatCommandArgs.h

```diff
@@ -99,6 +102,84 @@ struct ArgInfo<std::string, void>
     }
 };
 
+// enum
+template <typename T>
+struct ArgInfo<T, std::enable_if_t<std::is_enum_v<T>>>
+{
+    typedef typename std::underlying_type<T>::type uType;
+
+    static inline std::map<std::string, Optional<T>> MakeSearchMap()
+    {
+        std::map<std::string, Optional<T>> map;
+        for (T val : EnumUtils::Iterate<T>())
+        {
+            EnumText text = EnumUtils::ToString(val);
+
+            std::string title = std::string(text.Title);
```

Comment:
```suggestion
            std::string title(text.Title), constant(text.Constant);
```

Reply:
hell no. dont ever multiple variables in one line

Reply:
w/e fine, i mainly care about the `std::string title = std::string` being ugly